### ISSUE TITLE
Biome replacer not working when nature's spirit is presentt

### ISSUE THREAD

**Issue by JpnRndr01** [WerDei/Biome-Replacer#21]
How to reproduce:
Step 1: create a modpack with Natures Spirit, Biome Replacer and Terrablender (If forge make sure to get sinytra connector, forgified fabric api, connector extras and the fabric version of terrablender)
Step 2: Add "minecraft:cherry_grove > minecraft:taiga" to your biome_replaceer.properties
Step 2: Generate two worlds, with and without natures spirit
Step 3: Locate a cherry grove on the NS world and then go to the same coords on the non-NS world

heres my latest log in case it can help figure out why this is happening
https://mclo.gs/adISdSU

**Comment by WerDei** [comment]
This is a longstanding TerraBlender issue, and I'm not really sure how I can fix it. I [asked for help](https://github.com/Glitchfiend/TerraBlender/issues/188) over on TerraBlender's issue tracker, hopefully they'll point me in the right direction.

**Comment by JpnRndr01** [comment]
ah thanks, whats strange is that its only happening with NS for me... like BOP seems to work

**Comment by WerDei** [comment]
Since I linked this issue in a readme, here's a summary of the problem as I currently understand it:

This is most likely an issue on TerraBlender's side. TB divides the world into "regions", one for vanilla, and one for every mod that uses it. If a mod fails to provide the biome (say, it only adds cold biomes, but the region is in a warm part of the world), it will usually tell TB to place a vanilla biome there. 
In a perfect world, TB _should_ be using a data pack biome distribution for this, but unfortunately it does not. So you end up in a situation where half of biomes are replaced (the ones from the "vanilla" region), and another half aren't.

**Comment by optionalduck** [comment]
> Since I linked this issue in a readme, here's a summary of the problem as I currently understand it:
> 
> This is most likely an issue on TerraBlender's side. TB divides the world into "regions", one for vanilla, and one for every mod that uses it. If a mod fails to provide the biome (say, it only adds cold biomes, but the region is in a warm part of the world), it will usually tell TB to place a vanilla biome there. In a perfect world, TB _should_ be using a datapack provided biome distribution when this happens, but unfortunately it does not. So you end up in a situation where half of biomes are replaced (the ones from the "vanilla" region), and another half aren't.

Would it be possible to replace the biome with a vanilla biome and then replace it with the selected biome to replace?

**Comment by Vaulto350** [comment]
oh thank god its just a terrablender issue

i was worried i couldn't remove any BOP biomes, mainly the Fir ones, cause i have the New World mod that has fir trees and other stuff i like more

**Comment by Scarlet-Fury** [comment]
Can confirm, I tried to replace the deep dark with this mod as well and Terrablender regions still had it

Kinda frustrating and hoping compatiblity for it is fixed soon

But for now, the best solution I came up with when using Biome Replacer with Terrablender is stripping the deep dark of all skulk features using a KubeJS startup script and then disable structures to remove the Ancient Cities (Which also generate with Skulk)


Either that, or a mod that enables complete customization with world gen and compatibility with Terrablender would be SO helpful


I'm tempted to do it myself, but I'm still getting the hang of JavaScript, much less Java itself 😣



> > Since I linked this issue in a readme, here's a summary of the problem as I currently understand it:
> > 
> > This is most likely an issue on TerraBlender's side. TB divides the world into "regions", one for vanilla, and one for every mod that uses it. If a mod fails to provide the biome (say, it only adds cold biomes, but the region is in a warm part of the world), it will usually tell TB to place a vanilla biome there. In a perfect world, TB _should_ be using a datapack provided biome distribution when this happens, but unfortunately it does not. So you end up in a situation where half of biomes are replaced (the ones from the "vanilla" region), and another half aren't.
> 
> Would it be possible to replace the biome with a vanilla biome and then replace it with the selected biome to replace?
> Also I discovered replacing the deep dark biome only work when it generates under vanilla biome regions

**Comment by LegendarySpy** [comment]
> Kinda frustrating and hoping compatiblity for it is fixed soon

I don't want to hype up something that isn't 100% ready, but with the testing I've done the next update _should_ add support for TerraBlender via Biolith, as long as that is installed with TB you should be able to replace, remove or chanced replace. This should hopefully be out within the coming days.

**Comment by WerDei** [comment]
@Searaven114 Yes, the update is ready and functional, and I just need to find some time to polish things up and release it. Sorry for the delay!

**Comment by brassmeister** [comment]
Will this update be available for Forge on 1.20.1?

**Comment by YourMom2049** [comment]
> [@WerDei](https://github.com/WerDei) any news? For me I can replace most vanilla biomes at this point, except badlands, wooded badlands and eroded badlands, they get replaced in only half of the space.... Ive been frustrated with this shit for a solid year now because I want to use modded biomes but I have no intention of playing in a world with badlands or deserts existing...

I'm using only Biomes we've gone for additional biomes

**Comment by LegendarySpy** [comment]
Would you be able to continue this convo in the unoffical [discord](https://discord.com/invite/z3h4d3Ux3p), I can help you easier there, I compiled some test builds that I'd love some feedback on to help with polishing things up faster.
